qemu/tests/qemu_guest_agent.py

```diff
@@ -1850,6 +1850,181 @@ def ip_addr_check(session, mac_addr, ret_list, if_index, if_name):
+            )
+            for line in routes:
+                if line.startswith("default via"):
+                    line = line.replace("default via", "").strip()
+                    destination = "::"
+                    nexthop = line.split()[0]
+                    iface = line.split()[2]
+                    metric = line.split()[-3]
+                    desprefixlen = 0  # default desprefixlen should be 0
+                    processed_routes.append(
+                        {
+                            "iface": iface,
+                            "destination": destination,
+                            "desprefixlen": desprefixlen,
+                            "nexthop": nexthop,
+                            "metric": metric,
+                        }
+                    )
+                    continue
+                match = route_pattern.match(line)
+                if match:
+                    destination = match.group("destination")
+                    iface = match.group("iface")
+                    metric = match.group("metric")
+
+                    if "/" in destination:
+                        destination = destination.split("/")[0]
+                        desprefixlen = 64
+                    else:
+                        desprefixlen = 128
+                    nexthop = "::"
+                    processed_routes.append(
+                        {
+                            "iface": iface,
+                            "destination": destination,
+                            "desprefixlen": desprefixlen,
+                            "nexthop": nexthop,
+                            "metric": metric,
+                        }
+                    )
+            return processed_routes
+
+        session = self._get_session(params, self.vm)
+        self._open_session_list.append(session)
+
+        error_context.context("Getting route info via guest-agent API.", LOG_JOB.info)
+        qga_route_info = self.gagent.get_network_route()
+
+        error_context.context("Getting route info from inside the guest.", LOG_JOB.info)
+        ipv4_routes = parse_ipv4_route(session.cmd_output(self.params["cmd_ipv4route"]))
+        ipv6_routes = parse_ipv6_route(session.cmd_output(self.params["cmd_ipv6route"]))
+        guest_route_info = ipv4_routes + ipv6_routes
+
+        error_context.context(
+            "Compare the route info from qga and guest.", LOG_JOB.info
+        )
+        ipv4_keys = list(self.params["ipv4_keys"])
+        ipv6_keys = list(self.params["ipv6_keys"])
+
+        for guest_entry in guest_route_info:
+            matched_qga_entry = None
+            for qga_entry in qga_route_info:
+                if all(
+                    guest_entry.get(key) == qga_entry.get(key)
+                    for key in ["iface", "destination"]
+                ):
+                    matched_qga_entry = qga_entry
+                    break
```

Comment:
Hi @6-dehan As we discussed offline, please check these line again(`guest_entry` is a `str`).Thanks.

Reply:
As we discussed just right now, the guest_entry is type of 'dict', and the guest_route_info is type of 'list'. we got mistake before. so no problem at all. I'll update test result later.